tls/s2n_server_cert_verify.c

```diff
@@ -13,14 +13,123 @@
  * permissions and limitations under the License.
  */
 
+#include "tls/s2n_certificate_verify.h"
+#include "tls/s2n_connection.h"
+#include "crypto/s2n_hash.h"
+
+#include "stuffer/s2n_stuffer.h"
 #include "error/s2n_errno.h"
+#include "utils/s2n_safety.h"
+
+static int s2n_server_write_cert_verify_signature(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_server_generate_unsigned_cert_verify_content(struct s2n_connection *conn, struct s2n_stuffer *unsigned_content, struct s2n_hash_state *hash_copy, struct s2n_hash_state *message_hash);
+static uint8_t s2n_server_cert_verify_header_length();
 
 int s2n_server_cert_verify_send(struct s2n_connection *conn)
 {
-    S2N_ERROR(S2N_ERR_UNIMPLEMENTED);
+    struct s2n_stuffer *out = &conn->handshake.io;
+
+    /* Write the algorithm to out */
+    GUARD(s2n_stuffer_write_uint8(out, conn->secure.conn_hash_alg));
+    GUARD(s2n_stuffer_write_uint8(out, conn->secure.conn_sig_alg));
+
+    /* Write digital signature */
+    GUARD(s2n_server_write_cert_verify_signature(conn, out));
+
+    return 0;
 }
 
 int s2n_server_cert_verify_recv(struct s2n_connection *conn)
 {
-    S2N_ERROR(S2N_ERR_UNIMPLEMENTED);
+    struct s2n_stuffer *in = &conn->handshake.io;
+    DEFER_CLEANUP(struct s2n_blob signed_content = {0}, s2n_free);
+    DEFER_CLEANUP(struct s2n_stuffer unsigned_content ={0}, s2n_stuffer_free);
+    DEFER_CLEANUP(struct s2n_hash_state message_hash = {0}, s2n_hash_free);
+    DEFER_CLEANUP(struct s2n_hash_state hash_copy = {0}, s2n_hash_free);
```

Comment:
since `hash_copy` is only used in `s2n_server_generate_unsigned_cert_verify_content()`, would be better to move this into that function.